.github/workflows/pr.yml

```diff
@@ -31,3 +31,74 @@ jobs:
           ./mvnw ${MAVEN_ARGS} impsort:check --file pom.xml
       - name: Run unit tests
         run: ./mvnw ${MAVEN_ARGS} -B test --file pom.xml
+  spring-boot-integration-test:
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      matrix:
+        java: [ 11 ]
+        distribution: [ temurin ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Java and Maven
+        uses: actions/setup-java@v2
+        with:
+          distribution: ${{ matrix.distribution }}
+          java-version: ${{ matrix.java }}
+          cache: 'maven'
+      - name: Build
+        run: ./mvnw ${MAVEN_ARGS} clean install -DskipTests
+      - name: Kubernetes KinD Cluster
+        uses: container-tools/kind-action@v1
+        with:
+          version: v0.11.1
+          registry: true
+      - name: Install Cert-Manager
+        run: |
+          OS=$(go env GOOS); ARCH=$(go env GOARCH); curl -sSL -o kubectl-cert-manager.tar.gz https://github.com/cert-manager/cert-manager/releases/download/v1.7.2/kubectl-cert_manager-$OS-$ARCH.tar.gz
+          tar xzf kubectl-cert-manager.tar.gz
+          sudo mv kubectl-cert_manager /usr/local/bin
+          kubectl cert-manager x install
+      - name: Run Integration Test
+        run: |
+          set -x
```

Comment:
It would be nicer to have these action applied from java code, maybe part of junit (extension?) If that is not too much work. What do you think?

Reply:
This implies a lot of effort and I don't see much benefit as it is now (take into account that many of these steps could be automated in the future).